waspc/e2e-test/ShellCommands.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module ShellCommands
+  ( ShellCommand,
+    ShellCommandContext (..),
+    ShellCommandBuilder (..),
+    runShellCommandBuilder,
+    combineShellCommands,
+    cdIntoCurrentProject,
+    appendToWaspFile,
+    reformatPackageJson,
+    setDbToPSQL,
+    waspCliNew,
+    waspCliCompile,
+    waspCliMigrate,
+    waspCliBuild,
+  )
+where
+
+import Control.Monad.Reader (MonadReader (ask), Reader, runReader)
+import Data.List (intercalate)
+import System.FilePath (joinPath, (</>))
+
+-- NOTE: Should we consider separating shell parts, Wasp CLI parts, and test helper parts in the future?
+--       This would likely reqiure adoption of some library, and creating new layers of abstraction.
+--       Deemed not worth it right now by all.
+
+-- NOTE: Using `wasp-cli` herein so we can assume using latest `stack install` in CI and locally.
+
+-- TODO: In future, find a good way to test `wasp-cli start`.
+
+type ShellCommand = String
+
+-- Each shell command gets access to the current project name, and maybe other things in future.
+data ShellCommandContext = ShellCommandContext
+  { _ctxtCurrentProjectName :: String
```

Comment:
Interesting, didn't you have more stuff in the context a bit before?

Reply:
Yeah, I had the paths to the current and golden dirs in there, but turns out they were not used in shell commands (yet) so I removed them. But I think this place will grow over time if we add more complex shell commands 👍🏻